libparsec/crates/client/src/certif/mod.rs

```diff
@@ -51,6 +51,82 @@ use libparsec_types::prelude::*;
+// The base offset is the average network round trip time to compensate
+// the first round trip.
+
+// An optimization could be to measure the real round trip time for each request.
+// The issue with that is that we tackling two distinct issues
+// 1. the client clock is lagging behind
+// 2. the client connection is degraded
+// The current approach is only compensating for case 1.
+// Case 2 would benefit from a offset depending on the real round trip time,
+// but if we base our estimate of the round trip time on the expected timestamp
+// provided by the server, case 2 would benefit from an unjustified priority, as
+// it would pass for a client with a very degraded connection (within the limits of the ballpark though).
+// That issue would not arise if we monitor the round trip time only from the client point of view.
+
+/// This value is used to increment the timestamp provided by the backend
+/// when a manifest restamping is required. This value should be kept small
+/// compared to the certificate stamp ahead value, so the certificate updates have
+/// priority over manifest updates.
+/// # in microseconds, or 0.1 seconds, 100 ms (average network round trip time)
+pub const MANIFEST_STAMP_AHEAD_US: i64 = 100_000;
+
+/// This value is used to increment the timestamp provided by the backend
+/// when a realm certificate restamping is required. This value should be kept big
+/// compared to the manifest stamp ahead value, but less than USER_CERTIFICATE_STAMP_AHEAD_US
+///  so the certificate updates have priority over manifest updates.
+///  # microseconds, or 0.25 seconds
+pub const REALM_CERTIFICATE_STAMP_AHEAD_US: i64 = 250_000;
+
+/// This value is used to increment the timestamp provided by the backend
+/// when a user or device certificate restamping is required. This value should be kept big
+/// compared to the manifest stamp ahead value, so the certificate updates have
+/// priority over manifest updates.
+///  # microseconds, or 0.5 seconds
+pub const USER_CERTIFICATE_STAMP_AHEAD_US: i64 = 500_000;
+
+/// Archiving doesn't have to use a value as large 0.5 seconds
+/// # microseconds, or 1 milliseconds
+/// TODO: not used yet, see #6092
+pub const ARCHIVING_CERTIFICATE_STAMP_AHEAD_US: i64 = 1_000;
+
+pub enum GreaterTimestampOffset {
+    Manifest,
+    User,
+    Realm,
+    Archive,
+}
+
+impl From<GreaterTimestampOffset> for i64 {
+    fn from(value: GreaterTimestampOffset) -> Self {
+        match value {
+            GreaterTimestampOffset::Manifest => MANIFEST_STAMP_AHEAD_US,
+            GreaterTimestampOffset::User => USER_CERTIFICATE_STAMP_AHEAD_US,
+            GreaterTimestampOffset::Archive => ARCHIVING_CERTIFICATE_STAMP_AHEAD_US,
+            GreaterTimestampOffset::Realm => REALM_CERTIFICATE_STAMP_AHEAD_US,
+        }
+    }
+}
+
+/// ops to have the time provider
+/// offset depending on the type of certificate
+/// strictly_greater_than, the time expected by the server
```

Comment:
Minor suggestion to use structure from https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html

```suggestion
/// Return the timestamp to be used when the server requires a greater timestamp
///
/// The timestamp to be used will be whichever is greater between:
/// - now (obtained from `time_provider`)
/// - `strictly_greater_than` (expected by the server) + `offset` (depending on certificate type)
```